diff -Naur phpAdsNew/adclick.php /var/www/html/Awarez/phpAdsNew/adclick.php
--- phpAdsNew/adclick.php	2003-01-30 14:06:39.000000000 +0000
+++ /var/www/html/Awarez/phpAdsNew/adclick.php	2003-03-13 14:00:30.000000000 +0000
@@ -50,6 +50,8 @@
 
 if (!isset($bannerid) && isset($bannerID)) $bannerid = $bannerID;
 if (!isset($n)) $n = 'default';
+if (!isset($source)) $source = '';
+if ($source == '{derive}') $source = phpAds_deriveSource($source);
 
 
 // Fetch BannerID
@@ -109,8 +111,6 @@
 		
 		// If zoneid is not set, log it as a regular banner
 		if (!isset($zoneid)) $zoneid = 0;
-		if (!isset($source)) $source = '';
-		
 		
 		// Log clicks
 		if ($phpAds_config['block_adclicks'] == 0 ||
diff -Naur phpAdsNew/adframe.php /var/www/html/Awarez/phpAdsNew/adframe.php
--- phpAdsNew/adframe.php	2003-02-11 16:16:01.000000000 +0000
+++ /var/www/html/Awarez/phpAdsNew/adframe.php	2003-03-13 13:58:44.000000000 +0000
@@ -63,9 +63,10 @@
 if (!isset($what)) 		$what = '';
 if (!isset($clientid)) 	$clientid = 0;
 if (!isset($target)) 	$target = '_top';
-if (!isset($source)) 	$source = '';
 if (!isset($withtext)) 	$withtext = '';
 if (!isset($context)) 	$context = '';
+if (!isset($source))	$source = '';
+if ($source == '{derive}') $source = phpAds_deriveSource($source);
 
 // Remove referer, to be sure it doesn't cause problems with limitations
 if (isset($HTTP_SERVER_VARS['HTTP_REFERER'])) unset($HTTP_SERVER_VARS['HTTP_REFERER']);
diff -Naur phpAdsNew/adjs.php /var/www/html/Awarez/phpAdsNew/adjs.php
--- phpAdsNew/adjs.php	2003-02-11 14:09:07.000000000 +0000
+++ /var/www/html/Awarez/phpAdsNew/adjs.php	2003-03-12 17:14:53.000000000 +0000
@@ -93,9 +93,10 @@
 if (!isset($what)) 		$what = '';
 if (!isset($clientid)) 	$clientid = 0;
 if (!isset($target)) 	$target = '';
-if (!isset($source)) 	$source = '';
 if (!isset($withtext)) 	$withtext = '';
 if (!isset($context)) 	$context = '';
+if (!isset($source)) $source = '';
+if ($source == '{derive}') $source = phpAds_deriveSource($source);
 
 if (isset($exclude) && $exclude != '')
 {
diff -Naur phpAdsNew/adlayer.php /var/www/html/Awarez/phpAdsNew/adlayer.php
--- phpAdsNew/adlayer.php	2003-02-11 16:16:03.000000000 +0000
+++ /var/www/html/Awarez/phpAdsNew/adlayer.php	2003-03-12 17:15:44.000000000 +0000
@@ -163,9 +163,10 @@
 if (!isset($what)) $what = '';
 if (!isset($clientid)) $clientid = 0;
 if (!isset($target)) $target = '';
-if (!isset($source)) $source = '';
 if (!isset($withtext)) $withtext = '';
 if (!isset($context)) $context = '';
+if (!isset($source)) $source = '';
+if ($source == '{derive}') $source = phpAds_deriveSource($source);
 
 // Remove referer, to be sure it doesn't cause problems with limitations
 if (isset($HTTP_SERVER_VARS['HTTP_REFERER'])) unset($HTTP_SERVER_VARS['HTTP_REFERER']);
diff -Naur phpAdsNew/adlog.php /var/www/html/Awarez/phpAdsNew/adlog.php
--- phpAdsNew/adlog.php	2003-02-18 09:50:22.000000000 +0000
+++ /var/www/html/Awarez/phpAdsNew/adlog.php	2003-03-13 14:38:21.000000000 +0000
@@ -17,6 +17,10 @@
 // Figure out our location
 define ('phpAds_path', '.');
 
+// Start Timer
+require (phpAds_path."/classes/PageTimer.php");
+$phpAds_PageTimer = new PageTimer('adlog.php');
+$phpAds_PageTimer->start();
 
 
 /*********************************************************/
@@ -48,6 +52,7 @@
 if (isset($bannerid) && isset($clientid) && isset($zoneid))
 {
 	if (!isset($source)) $source = '';
+	if ($source == '{derive}') $source = phpAds_deriveSource($source);
 	
 	if ($phpAds_config['block_adviews'] == 0 ||
 	   ($phpAds_config['block_adviews'] > 0 && 
@@ -104,4 +109,6 @@
      chr(0x01).chr(0x00).chr(0x01).chr(0x00).chr(0x00).chr(0x02).chr(0x02).chr(0x44).
      chr(0x01).chr(0x00).chr(0x3B);
 
+$phpAds_PageTimer->stop();
+$phpAds_PageTimer->log();
 ?>
\ No newline at end of file
diff -Naur phpAdsNew/admin/stats-global-client.php /var/www/html/Awarez/phpAdsNew/admin/stats-global-client.php
--- phpAdsNew/admin/stats-global-client.php	2003-02-14 14:36:43.000000000 +0000
+++ /var/www/html/Awarez/phpAdsNew/admin/stats-global-client.php	2003-03-13 14:43:36.000000000 +0000
@@ -18,6 +18,11 @@
 require ("config.php");
 require ("lib-statistics.inc.php");
 
+// Start Timer
+require (phpAds_path.'/classes/PageTimer.php');
+$phpAds_PageTimer = new PageTimer('stats-global-client.php');
+$phpAds_PageTimer->start();
+
 
 // Register input variables
 phpAds_registerGlobal ('expand', 'collapse', 'hideinactive', 'listorder', 'orderdirection');
@@ -685,4 +690,8 @@
 
 phpAds_PageFooter();
 
+// Stop Timer
+$phpAds_PageTimer->stop();
+$phpAds_PageTimer->log();
+
 ?>
\ No newline at end of file
diff -Naur phpAdsNew/adpopup.php /var/www/html/Awarez/phpAdsNew/adpopup.php
--- phpAdsNew/adpopup.php	2003-02-11 16:16:03.000000000 +0000
+++ /var/www/html/Awarez/phpAdsNew/adpopup.php	2003-03-12 17:18:30.000000000 +0000
@@ -148,9 +148,10 @@
 if (!isset($what)) 		$what = '';
 if (!isset($clientid)) 	$clientid = 0;
 if (!isset($target)) 	$target = '_new';
-if (!isset($source)) 	$source = '';
 if (!isset($withtext)) 	$withtext = '';
 if (!isset($context)) 	$context = '';
+if (!isset($source)) $source = '';
+if ($source == '{derive}') $source = phpAds_deriveSource($source);
 
 // Remove referer, to be sure it doesn't cause problems with limitations
 if (isset($HTTP_SERVER_VARS['HTTP_REFERER'])) unset($HTTP_SERVER_VARS['HTTP_REFERER']);
diff -Naur phpAdsNew/adview.php /var/www/html/Awarez/phpAdsNew/adview.php
--- phpAdsNew/adview.php	2003-02-18 09:50:22.000000000 +0000
+++ /var/www/html/Awarez/phpAdsNew/adview.php	2003-03-12 17:19:37.000000000 +0000
@@ -61,8 +61,9 @@
 if (isset($clientID) && !isset($clientid)) $clientid = $clientID;
 if (!isset($clientid)) $clientid = 0;
 if (!isset($what)) $what = '';
-if (!isset($source)) $source = '';
 if (!isset($n)) $n = 'default';
+if (!isset($source)) $source = '';
+if ($source == '{derive}') $source = phpAds_deriveSource($source);
 
 // Remove referer, to be sure it doesn't cause problems with limitations
 if (isset($HTTP_SERVER_VARS['HTTP_REFERER'])) unset($HTTP_SERVER_VARS['HTTP_REFERER']);
diff -Naur phpAdsNew/config.inc.php /var/www/html/Awarez/phpAdsNew/config.inc.php
--- phpAdsNew/config.inc.php	2003-01-30 14:06:40.000000000 +0000
+++ /var/www/html/Awarez/phpAdsNew/config.inc.php	2003-03-12 11:42:48.000000000 +0000
@@ -1,10 +1,5 @@
 <?php
 
-// IMPORTANT! DO NOT MANUALLY EDIT THIS FILE! 
-// IF YOU WANT TO INSTALL PHPADSNEW PLEASE FOLLOW 
-// THE INSTRUCTIONS IN THE ADMINISTRATOR GUIDE.
-
-
 /*********************************************************/
 /* Database configuration                                */
 /*********************************************************/
@@ -13,16 +8,16 @@
 $phpAds_config['dbhost'] = 'localhost';
 
 // Database port
-$phpAds_config['dbport'] = '3306';
+$phpAds_config['dbport'] = 3306;
 
 // Database username
-$phpAds_config['dbuser'] = '';
+$phpAds_config['dbuser'] = 'phpadsnew';
 
 // Database password
-$phpAds_config['dbpassword'] = '';
+$phpAds_config['dbpassword'] = 'buckyball';
 
 // Database name
-$phpAds_config['dbname'] = '';
+$phpAds_config['dbname'] = 'phpadsnew';
 
 // Database table names
 $phpAds_config['tbl_adclicks'] = 'phpads_adclicks';
@@ -38,13 +33,13 @@
 $phpAds_config['tbl_images'] = 'phpads_images';
 $phpAds_config['tbl_userlog'] = 'phpads_userlog';
 $phpAds_config['tbl_cache'] = 'phpads_cache';
-$phpAds_config['tbl_targetstats'] = 'tbl_targetstats';
+$phpAds_config['tbl_targetstats'] = 'phpads_targetstats';
 
 // Database table name prefix
 $phpAds_config['table_prefix'] = 'phpads_';
 
 // Database table type
-$phpAds_config['table_type'] = 'MYISAM';
+$phpAds_config['table_type'] = 'INNODB';
 
 // Use persistent connections to the database
 $phpAds_config['persistent_connections'] = false;
@@ -63,11 +58,7 @@
 /*********************************************************/
 
 // The URL to your phpAds-installation
-$phpAds_config['url_prefix'] = 'http://www.your-url.com/phpAdsNew';
-
-// The Language of the Interface (eg. english, german, spanish) 
-// for the right string look for the directory name in the language subdirectory
-$phpAds_config['language'] = 'english';
+$phpAds_config['url_prefix'] = 'http://10.20.40.5/Awarez/phpAdsNew';
 
 // Is the admin interface enabled
 $phpAds_config['ui_enabled'] = true;
@@ -92,7 +83,7 @@
 // 1 = IP2Country
 // 2 = GeoIP
 // 3 = mod_geoip
-$phpAds_config['geotracking_type'] = 0;
+$phpAds_config['geotracking_type'] = '';
 
 // Location of the geotracking database
 $phpAds_config['geotracking_location'] = '';
@@ -110,7 +101,7 @@
 /*********************************************************/
 
 // Use compact or verbose statistics
-$phpAds_config['compact_stats'] = true;
+$phpAds_config['compact_stats'] = false;
 
 // Enabled logging of adviews?
 $phpAds_config['log_adviews'] = true;
@@ -125,13 +116,13 @@
 $phpAds_config['log_hostname'] = true;
 
 // Log only the IP address even if a hostname is available
-$phpAds_config['log_iponly'] = true;
+$phpAds_config['log_iponly'] = false;
 
 // Use beacons to log adviews
 $phpAds_config['log_beacon'] = true;
 
 // Hosts to ignore (don't count adviews coming from them)
-$phpAds_config['ignore_hosts'] = array ();   // Example: array('slashdot.org', 'microsoft.com');
+$phpAds_config['ignore_hosts'] = array ();
 
 // Block logging of views for xx seconds after the last entry
 // This is to prevent logging after each page reload
@@ -164,7 +155,7 @@
 
 // Policy file location
 // For example:
-// $phpAds_config['p3p_policy_location'] = 'http://www.your-url.com/w3c/p3p.xml';
+// $phpAds_config['p3p_policy_location'] = '';
 $phpAds_config['p3p_policy_location'] = '';
 
 
@@ -175,7 +166,7 @@
 
 // Delivery caching type?
 // Possible options: none, db, file or shm
-$phpAds_config['delivery_caching'] = 'db';
+$phpAds_config['delivery_caching'] = 'file';
 
 // Use conditional keywords?
 $phpAds_config['con_key'] = true;
@@ -212,12 +203,11 @@
 
 
 
-
 /*********************************************************/
 /* phpAdsNew self configuration code - don't change      */
 /*********************************************************/
 
-define('phpAds_installed', false);
+define('phpAds_installed', true);
 
 // Disable magic_quotes_runtime
 set_magic_quotes_runtime(0);
diff -Naur phpAdsNew/libraries/CVS/Entries /var/www/html/Awarez/phpAdsNew/libraries/CVS/Entries
--- phpAdsNew/libraries/CVS/Entries	2003-03-18 21:28:08.000000000 +0000
+++ /var/www/html/Awarez/phpAdsNew/libraries/CVS/Entries	2003-03-12 20:23:58.000000000 +0000
@@ -18,4 +18,8 @@
 /lib-warnings.inc.php/2.0/Tue Oct 29 21:07:47 2002//
 /lib-xmlrpc.inc.php/2.0/Tue Oct 29 21:07:47 2002//
 /lib-xmlrpcs.inc.php/2.0/Tue Oct 29 21:07:48 2002//
-D
+D/defaults////
+D/deliverycache////
+D/geotargeting////
+D/layerstyles////
+D/resources////
diff -Naur phpAdsNew/libraries/CVS/Entries.Log /var/www/html/Awarez/phpAdsNew/libraries/CVS/Entries.Log
--- phpAdsNew/libraries/CVS/Entries.Log	2003-03-18 21:28:09.000000000 +0000
+++ /var/www/html/Awarez/phpAdsNew/libraries/CVS/Entries.Log	1970-01-01 01:00:00.000000000 +0100
@@ -1,5 +0,0 @@
-A D/defaults////
-A D/deliverycache////
-A D/geotargeting////
-A D/layerstyles////
-A D/resources////
diff -Naur phpAdsNew/libraries/geotargeting/geo-mod_geoip.inc.php /var/www/html/Awarez/phpAdsNew/libraries/geotargeting/geo-mod_geoip.inc.php
--- phpAdsNew/libraries/geotargeting/geo-mod_geoip.inc.php	2003-02-18 09:50:23.000000000 +0000
+++ /var/www/html/Awarez/phpAdsNew/libraries/geotargeting/geo-mod_geoip.inc.php	2003-03-12 17:22:54.000000000 +0000
@@ -33,7 +33,7 @@
 {
 	// $addr and $db parameter is ignored and is here for API consistency only
 	
-	$country = @apache_note('GEOIP_COUNTRY_CODE');
+	$country = $HTTP_SERVER_VARS['GEOIP_COUNTRY_CODE'];
 	
 	if ($country != '' && $country != '--')
 	{
diff -Naur phpAdsNew/libraries/lib-io.inc.php /var/www/html/Awarez/phpAdsNew/libraries/lib-io.inc.php
--- phpAdsNew/libraries/lib-io.inc.php	2003-01-16 08:58:18.000000000 +0000
+++ /var/www/html/Awarez/phpAdsNew/libraries/lib-io.inc.php	2003-03-14 12:19:28.000000000 +0000
@@ -44,6 +44,100 @@
 	}
 }
 
+function phpAds_deriveSource ($str)
+{
+	global $HTTP_SERVER_VARS, $phpAds_config;
+
+	$qs = array();
+	$path = array();
+		
+	// Break down the referer...
+	if (isset($HTTP_SERVER_VARS['HTTP_REFERER']))
+	{
+		$referer = parse_url(strtolower($HTTP_SERVER_VARS['HTTP_REFERER']));
+		if (isset($referer['host']))
+		{
+			$host = $referer['host'];
+			$domains = array_reverse( explode(".", $host) );
+		}
+		else
+		{
+			$host = '';
+			$domains = array();  //empty array
+		}
+			
+		if (isset($referer['path']))
+		{
+			$path = '';
+			if ( substr_count($referer['path'], '/') > 1)
+				$path = substr($referer['path'], 1, strrpos($referer['path'], "/") - 1 );
+			$dirs = explode("/", $path);
+		}
+		else
+		{
+			$path = '';
+			$dirs = array();
+		}
+		
+		if (isset($referer['query']))
+		{
+			$tmpqs = explode("&", $referer['query']);
+			if ( is_array($tmpqs) )
+			{
+				foreach ( $tmpqs as $value )
+				{
+					$tmparg = strtok( $value, "=" );
+					$qs[$tmparg] = strtok('');
+				}
+			}
+		}
+		else
+			$qs = array();
+
+		// Initially, hard code the 'rules' for determining the source.
+		// Ebookers Rules...
+		if (preg_match('#ebookers|hoteldiscounts|interhome#i', $host) )  //Ebookers Rules
+		{
+			$source = 'ebookersfr';
+			if (preg_match('#hoteldiscounts#i', $host) )
+			{
+				$path = 'hoteldiscounts';
+			}
+			elseif (preg_match('#interhome#i', $host) )
+			{
+				$path = 'locations';
+			}
+			if (strlen($path) > 0)
+				$source = $source.'/'.$path;
+		}
+		elseif (preg_match('#ticketmaster#i', $host) ) //Ticketmaster Rules
+		{
+			$source = 'ticketmaster';
+			if (strlen($qs['category']) > 0)
+			{
+				$source = $source.'/'.$qs['category'];
+				if (strlen($qs['minorname']) > 0)
+					$source = $source.'/'.$qs['minorname'];
+			}
+		}
+		elseif (preg_match('#deliaonline#i', $host) ) // Delia Online
+		{
+			$source = 'deliaonline';
+			if (strlen($path) > 0)
+				$source = $source.'/'.$path;
+		}
+		else
+		{
+			$source = $host;
+			if (strlen($path) > 0)
+				$source = $source.'/'.$path;
+		}
+		
+		return $source;
+	}
+
+	return $str;
+}
 
 /*********************************************************/
 /* Recursive add slashes to an array                     */
